docs/guide.md

```diff
@@ -0,0 +1,210 @@
+# Getting Started with PatternPack
```

Comment:
We need to define tone/voice for documentation.  Personally I don't like to use "You" in this type of writing as most of the work is not owned by the individual authoring it.